GVFS/GVFS.Common/Http/HttpRequestor.cs

```diff
@@ -249,10 +306,63 @@ private static string GetSingleHeaderOrEmpty(HttpHeaders headers, string headerN
             IEnumerable<string> values;
             if (headers.TryGetValues(headerName, out values))
             {
-                return values.First();                
+                return values.First();
             }
 
             return string.Empty;
         }
+
+        private string LoadCertificatePassword(string certId, GitProcess git)
+        {
+            if (git.TryGetCertificatePassword(this.Tracer, certId, out var password, out var error))
+            {
+                return password;
+            }
+
+            return null;

```

Comment:
We will try to load the certificate without the password. If that fails, we'll return null for certificate, which means, that we'll try to access our HTTP endpoint without it, which will likely result in an authentication error. 
If there was an error, while retrieving password, it will be logged. 

Another way would be to stop the flow immediately, if either password or certificate can not be retrieved, but I think allowing the customer to proceed till he actually fails a request due to the missing certificate is less disruptive. 

One thing we should probably do, is to log a warning, stating that password was configured, but not retrieved (and same for certificate).

Reply:
Added the warning messages